### PR TITLE
btree updates

### PR DESCRIPTION
**Pull request by seebs**
## Overview

Noticed in passing that a lot of the runtime of value-related stuff is living in the code that finds things in the btrees, and there were some old TODO comments that suggested possible opportunities. Also looked at upstream a bit, and ended up sending them a benchmark justifying our local change to the `overflow` function, then decided to migrate the tests from that package back into our tree so we can be testing the btree code more carefully.

Also adjusted the tuning parameters, and inlined the sense of the comparison that we were using a cmp function for, which turns out to make btree operations themselves 10-30% faster.

No documentation changes, this shouldn't be visible to users.

I probably need to rebase this to master, but I wanted to get the actual changes looked at. (The btree_test code is probably less urgent to check -- if the tests pass now, I *probably* translated them correctly from the upstream form, which relied on both K and V being `interface{}`.)

## Pull request checklist

- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [X] I have included tests that cover my changes.
- [X] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by seebs** [comment]
i have no idea at all how i ended up with this looking like a 60+ patch set. fixed it.

File: roaring/btree.go [hunk 1]
```diff
@@ -379,27 +374,29 @@ func (t *tree) find(q interface{}, k uint64) (i int, ok bool) {
 		for l <= h {
 			m := (l + h) >> 1
 			mk = x.x[m].k
-			switch cmp := t.cmp(k, mk); {
-			case cmp > 0:
+			// note semicolon; the condition is the unspecified implicit true
```
Comment (jaffee) [inline roaring/btree.go]: looks like this comment shouldn't be here any more

Comment (seebs) [reply roaring/btree.go]: yeah, that was from an intermediate form where i had inlined the comparison, so it was `switch cmp := k - mk; {`

**Comment by jaffee** [comment]
merge?

**Pull request merged**